java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1495,6 +1495,44 @@ private static native void cudfColumnViewStrings(long cudfColumnHandle, long dat
                                                    long deviceOutputDataPtr,
                                                    int numRows, int dtype, int nullCount);
 
+  /**
+   * Native method to switch all characters in a column of strings to lowercase characters.
+   * @param cudfColumnHandle native handle of the gdf_column being operated on.
+   * @return native handle of the resulting cudf column, used to construct the Java column
+   *         by the lower method.
+   */
+  private static native long lowerStrings(long cudfColumnHandle);
+
+  /**
+   * Native method to switch all characters in a column of strings to uppercase characters.
+   * @param cudfColumnHandle native handle of the gdf_column being operated on.
+   * @return native handle of the resulting cudf column, used to construct the Java column
+   *         by the upper method.
+   */
+  private static native long upperStrings(long cudfColumnHandle);
+
+  /**
+   * Wrap static native string capitilization methods, retrieves the column vectors cudf native
+   * handle and uses it to invoke the native function that calls NVStrings' upper method. Does
+   * support String categories yet.
+   * @return A new ColumnVector containing the upper case versions of the strings in the original
+   *         column vector.
+   */
+  public ColumnVector upper() {
+    return new ColumnVector(upperStrings(getNativeCudfColumnAddress()));
```

Comment:
could we add assertions that the type of this column is a string?

Reply:
thanks for the change @rwlee, it would be better if you added a reason for the assertion. We are not consistent with this, but it's nice to get a message:
`assert type == DType.STRING : "A column of type string is required for .lower() operation"`